### ISSUE TITLE
C/C++ plugin crashes various operations in TypeScript (and Javascript) code

### ISSUE THREAD

**Issue by theovosse** [microsoft/vscode-cpptools#238]
```
VSCode Version: 1.5.2
OS Version: 10.11.6
```

Steps to Reproduce:

```
Open typscript project
type cmd-T and start typing the name of a symbol
```

This resulted in a crash in Microsoft.VSCode.CPP.Extension.darwin here:

> ```
> 0 Microsoft.VSCode.CPP.Extension.darwin 0x000000010acfc264 code_store::a_store::get_structure_version() + 4
> 1 Microsoft.VSCode.CPP.Extension.darwin 0x000000010acdaf01 browse_engine::populate_filename_cache() + 49
> 2 Microsoft.VSCode.CPP.Extension.darwin 0x000000010acdcb61 browse_engine::query_files(char const, std::__1::vector >&) + 129
> 3 Microsoft.VSCode.CPP.Extension.darwin 0x000000010aca1426 message_handler::workspace_symbol(vscode::WorkspaceSymbolParams) + 70
> 4 Microsoft.VSCode.CPP.Extension.darwin 0x000000010ac9a615 vscode::handler_base::dispatch(vscode::vscode_client_message&&, vscode::vscode_server_message&, bool) + 7637
> 5 Microsoft.VSCode.CPP.Extension.darwin 0x000000010ac98559 vscode::handler_base::handle_message(vscode::vscode_client_message&&) + 57
> 6 Microsoft.VSCode.CPP.Extension.darwin 0x000000010ac9849b vscode::handler_base::main_loop()::'lambda'()::operator()() const + 75
> 7 Microsoft.VSCode.CPP.Extension.darwin 0x000000010ac98410 void std::__1::__thread_proxystd::__1::tuple<vscode::handler_base<message_handler::main_loop()::'lambda'()> >(void*) + 96
> 8 libsystem_pthread.dylib 0x00007fff955f899d _pthread_body + 131
> 9 libsystem_pthread.dylib 0x00007fff955f891a _pthread_start + 168
> 10 libsystem_pthread.dylib 0x00007fff955f6351 thread_start + 13
> ```

Removing the plugin restored normal functioning.

**Comment by sean-mcmanus** [comment]
I am able to repro this. The crash happens the 1st time a folder is opened.

**Comment by sean-mcmanus** [comment]
This is the same as https://github.com/Microsoft/vscode-cpptools/issues/236 but they have different, important repro details, so keeping both open seems fine to me.

**Comment by sean-mcmanus** [comment]
The bug repros on all OS's. And it only repros after entering the 3rd character (otherwise our symbol processing code is skipped).

**Comment by sean-mcmanus** [comment]
This has been fixed. It'll be in a future release.

**Comment by sean-mcmanus** [comment]
The fix didn't make 0.9.2, but it should be in the next version.

**Comment by theovosse** [comment]
Thanks for the heads-up.
